src/torch/layer.ml

```diff
@@ -64,6 +68,58 @@ let linear vs ?activation ?(use_bias = true) ?w_init ~input_dim output_dim =
   { apply }
 ;;
 
+let conv1d
+      vs
+      ~ksize:k1
+      ~stride
+      ?activation
+      ?(use_bias = true)
+      ?w_init
+      ?(padding = 0)
+      ?(groups = 1)
+      ~input_dim
+      output_dim
+  =
+  let w =
+    let shape = [ output_dim; input_dim / groups; k1 ] in
+    match w_init with
+    | None -> kaiming_uniform vs ~shape ~a:(Float.sqrt 5.) ~name:"weight"
+    | Some init -> Var_store.new_var vs ~shape ~init ~name:"weight"
+  in
+  let b =
+    if use_bias
+    then Some (Var_store.new_var vs ~shape:[ output_dim ] ~init:Zeros ~name:"bias")
+    else None
+  in
+  let apply xs = Tensor.conv1d xs w b ~padding ~stride ~groups |> apply ?activation in
```

Comment:
various errors with the assumed function signature here; this doesn't compile as-is